cluster-operator/src/main/java/io/strimzi/operator/cluster/Main.java

```diff
@@ -185,39 +188,48 @@ private static KafkaConnectS2IAssemblyOperator createS2iOperator(Vertx vertx, Ku
         kafkaConnectS2iCrdOperator = new CrdOperator<>(vertx, osClient, KafkaConnectS2I.class, KafkaConnectS2IList.class, DoneableKafkaConnectS2I.class);
         networkPolicyOperator = new NetworkPolicyOperator(vertx, client);
         podDisruptionBudgetOperator = new PodDisruptionBudgetOperator(vertx, client);
-        kafkaConnectS2IClusterOperations = new KafkaConnectS2IAssemblyOperator(vertx, isOpenShift,
+        kafkaConnectS2IClusterOperations = new KafkaConnectS2IAssemblyOperator(vertx, pfa,
                 certManager,
                 kafkaConnectS2iCrdOperator,
                  configMapOperations, deploymentConfigOperations,
                 serviceOperations, imagesStreamOperations, buildConfigOperations, secretOperations, networkPolicyOperator, podDisruptionBudgetOperator, resourceOperatorSupplier, versions, imagePullPolicy);
         return kafkaConnectS2IClusterOperations;
     }
 
-    static Future<Boolean> isOnOpenShift(Vertx vertx, KubernetesClient client, ClusterOperatorConfig config)  {
-        if (config.isAssumeOpenShift() != null)  {
-            log.debug("OpenShift has been set to {} through {}.", config.isAssumeOpenShift(), ClusterOperatorConfig.STRIMZI_ASSUME_OPENSHIFT);
-            return Future.succeededFuture(config.isAssumeOpenShift());
-        } else if (client.isAdaptable(OkHttpClient.class)) {
+    static Future<PlatformFeaturesAvailability> getPlatformFeaturesAvailability(Vertx vertx, KubernetesClient client, ClusterOperatorConfig config)  {
```

Comment:
Just a thought ... feel free to ignore it. But I would maybe split this into two sub-functions - one to detect  OpenShift and one to get the version.